pkg/sync/sync.go

```diff
@@ -19,6 +19,53 @@ type SyncClient interface {
 	ExtractProjectToComponent(common.ComponentInfo, string, io.Reader) error
 }
 
+// GetTarReader creates a tar file from files and return a reader to it
+func GetTarReader(localPath string, targetPath string, copyFiles []string, globExps []string, copyBytes map[string][]byte) (reader io.Reader, err error) {
+	// Destination is set to "ToSlash" as all containers being ran within OpenShift / S2I are all
+	// Linux based and thus: "\opt\app-root\src" would not work correctly.
+	dest := filepath.ToSlash(filepath.Join(targetPath, filepath.Base(localPath)))
+
+	klog.V(4).Infof("CopyFile arguments: localPath %s, dest %s, targetPath %s, copyFiles %s, globalExps %s", localPath, dest, targetPath, copyFiles, globExps)
+	reader, writer := io.Pipe()
+	// inspired from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cp.go#L235
+	go func() {
+		defer writer.Close()
+
+		tarWriter := tar.NewWriter(writer)
+
+		err := makeTar(localPath, dest, copyFiles, globExps, tarWriter)
+		if err != nil {
+			log.Errorf("Error while creating tar: %#v", err)
+			os.Exit(1)
+		}
+
+		// For each of the files passed, write them to the tar.Writer
+		// No files can be passed, but if any are, they will be bundled up in the
+		// tar as well.
+		for name, content := range copyBytes {
+			hdr := &tar.Header{
+				Name: name,
+				Mode: 7770,
```

Comment:
We might need approval that this is acceptable permission too?

Reply:
I tested with `421` rather than `7770` and it didn't fail, so we could change it if you wanted to be more secure.

Reply:
yeah maybe that would be better

Reply:
you wanna raise a PR?

Reply:
Sure yeah